### PR TITLE
Arconomy: Balance Update 2023

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request
Hey everyone I've been dead and tired and distracted and wage cage'd and hopefully you'll see why in the coming weeks but this isn't that PR this is something slightly different

So! Changes in this Pull Request:

- I've implemented the changes in #76999 but with the commented suggestion so that now, mechanically, the foreign species modifier is now considered to be the floor payment amount. This means a functional buff to human paychecks, but in game it's still represented as a non-human nerf.

- The inflation mechanic on vending machines has been removed. This means that vending machines will keep a consistent cost at all times. Inflation as a mechanic is still utilized for mail, as well as kept on the SS level in order to ensure that the market crash events still works properly.
- Market Crash event has been improved, with some new flavor messages added, as well as using a new formula where inflation increases for the period of the event along a 5.5*log(time) scale.

- Tweaks have been made to the accounting computer TGUI to better follow standard conventions and not use a labeledList. It also properly shows the player facing paycheck modifier as well. It also looks a bit better now!

## Why It's Good For The Game

Justification for the change on the 90% paycheck modifier change has been discussed in #76999 already, but going with the more positive spin angle.

The inflation mechanic in general has rather unintuitive for a number of reasons, but the biggest one being that the mechanic doesn't really have any bearing that players in-game can really effect. Especially with passive paychecks, the only way to work around inflation is to make... other players spend their money. That's hardly an easy task with a 10 player crew, let alone a 90 player crew. Plus, the outcome is that it... punishes players for engaging with the economy if done correctly? Point being it was a decent idea but it doesn't mash with the desired outcomes of the player economy in it's current state.

Improvements to the accounting TGUI showcase were necessary as a result, and just look a bit nicer (I will get a screenshot in a moment). There is still room to potentially look into command staff being able to give raises/pay-cuts to crew but I'd want to treat that similar to opening up crew slots, so I'm putting that off for now.

## Changelog

:cl:
qol: Improvements to the accounting console in the HOP office.
balance: Non-human species now receive a 90% racial paycut, up from 75%.
balance: The inflation mechanic affecting vending machines has been removed, except during the market crash event.
balance: The market crash event now stands as it's own separate event, and ramps up the cost of vendor items until finally popping and returning to normal. Market crash now lasts longer to have more impact while it scales with time.
/:cl:

### CODE REVIEW

**Comment by FernandoJ8** [comment]
It's actually a 90.9% racial paycut (nerd emoji)

**Comment by atlasle** [comment]
can you make vending machines always have the same prices for things instead of basing it off the paychecks, so people can remember them and it creates a baseline so people can understand what money is worth

**Comment by atlasle** [comment]
I know but can't paychecks be altered by random events in the round? would these alter vending machine prices then?

**Comment by ArcaneMusic** [comment]
> I know but can't paychecks be altered by random events in the round? would these alter vending machine prices then?

Paychecks and payscale are not altered by random events,  so those do not alter vending machine prices. The market crash event alters vendor prices temporarily, but not paychecks.

**Comment by atlasle** [comment]
alright thank you very much this pr is great

File: code/controllers/subsystem/economy.dm [hunk 1]
```diff
@@ -228,6 +208,29 @@ SUBSYSTEM_DEF(economy)
 		"vendor" = vendor,
 	))
 
+/**
+ * Iterates over the machines list for vending machines, resets their regular and premium product prices (Not contraband), and sends a message to the newscaster network.
+ */
+/datum/controller/subsystem/economy/proc/update_vending_prices()
+	var/list/obj/machinery/vending/prices_to_update = list()
+		// Assoc list of "z level" -> if it's on the station
+		// Hack, is station z level is too expensive to do for each machine, I hate this place
```
Comment (Jacquerel) [inline code/controllers/subsystem/economy.dm:217]:
```suggestion
	// Assoc list of "z level" -> if it's on the station
	// Hack, is station z level is too expensive to do for each machine, I hate this place
```

File: code/modules/events/market_crash.dm [hunk 2]
```diff
@@ -11,33 +11,46 @@
 	description = "Temporarily increases the prices of vending machines."
 
 /datum/round_event/market_crash
-	var/market_dip = 0
+	var/every_fifth = 1
```
Comment (Jacquerel) [inline code/modules/events/market_crash.dm:14]: Can you document this var? It's not very self-explanatory.

File: code/modules/events/market_crash.dm [hunk 3]
```diff
@@ -11,33 +11,46 @@
...
 	SSeconomy.price_update()
+	SSeconomy.update_vending_prices()
 	priority_announce("Prices for on-station vendors have now stabilized.", "Nanotrasen Accounting Division")
 
+/datum/round_event/market_crash/tick()
+	. = ..()
+	every_fifth = every_fifth + 1
+	SSeconomy.inflation_value = 5.5*(log(activeFor+1))
+	if(every_fifth == 5)
+		every_fifth = 1
```
Comment (Jacquerel) [inline code/modules/events/market_crash.dm:55]: If it starts at 1 and adds 1 every time, doesn't this activate every fourth tick?

File: code/modules/events/market_crash.dm [hunk 4]
```diff
@@ -11,33 +11,46 @@
...
 	. = ..()
-	SSeconomy.station_target += market_dip
 	REMOVE_TRAIT(SSeconomy, TRAIT_MARKET_CRASHING, MARKET_CRASH_EVENT_TRAIT)
 	SSeconomy.price_update()
+	SSeconomy.update_vending_prices()
 	priority_announce("Prices for on-station vendors have now stabilized.", "Nanotrasen Accounting Division")
 
+/datum/round_event/market_crash/tick()
+	. = ..()
+	every_fifth = every_fifth + 1
```
Comment (Jacquerel) [inline code/modules/events/market_crash.dm:52]: Not an important one but why `x = x + 1` instead of `x += 1` or `x++` ?

**Review by Jacquerel — CHANGES_REQUESTED** [review]
Huh it almost seems like the word "economy" has turned off other maintainers from reviewing this 🤔 
Anyway there's actually not much to comment on here, pretty much looks good to me aside from that I am not sure that your tick fires as advertised.

File: code/modules/events/market_crash.dm (hunk 2 above)
Comment (Jacquerel) [reply code/modules/events/market_crash.dm:14]: Maybe also just rename it to something like "tick counter" and make a define for how many times we want it to tick before doing our thing tbh.

File: code/modules/events/market_crash.dm (hunk 4 above)
Comment (ArcaneMusic) [reply code/modules/events/market_crash.dm:52]: Occasionally I'll do that for extreme certainty, but I'll swap that over to ++, no reason not to

File: code/modules/events/market_crash.dm (hunk 3 above)
Comment (ArcaneMusic) [reply code/modules/events/market_crash.dm:55]: Doh

**Pull request merged**